### ISSUE TITLE
[Bug] - Deleted Sessions back by server pools stuck in 'user_delete_pending' or 'admin_delete_pending'

### ISSUE THREAD

**Issue by frznfngrs** [kasmtech/workspaces-issues#425]
### Existing Resources

- [x] Please search the existing issues for related problems
- [x] Consult the product documentation : [Docs](https://kasmweb.com/docs/latest/index.html)
- [x] Consult the FAQ : [FAQ](https://kasmweb.com/docs/latest/FAQ.html)
- [x] Consult the Troubleshooting Guide : [Guide](https://kasmweb.com/docs/latest/guide/troubleshooting.html)
- [x] Reviewed existing training videos: [Youtube](https://www.youtube.com/@KasmWorkspaces )

**Describe the bug**
 Deleted Sessions back by server pools stuck in 'user_delete_pending' or 'admin_delete_pending'.
Manual removal of sessions from db possible and clears them.  

**To Reproduce**
1. Create Session using Server RDP pool backed workspace
2. delete session
3. View Admin Live sessions
4. See sessions remain stuck in 'user_delete_pending' or if session deleted by admin session stuck or 'admin_delete_pending'

**Expected behavior**
Sessions should delete.

**Screenshots**
<img width="659" alt="image" src="https://github.com/kasmtech/workspaces-issues/assets/19555532/daab8b0d-92eb-4198-a798-f13a53037715">
<img width="644" alt="image" src="https://github.com/kasmtech/workspaces-issues/assets/19555532/73a1e27f-42e0-417f-8262-cfc559770532">
<img width="952" alt="image" src="https://github.com/kasmtech/workspaces-issues/assets/19555532/84794aab-9ed7-4aa4-8e15-aa5ecb7f65e3">

**Workspaces Version**
e.g Version 1.13.1

**Workspaces Installation Method**
Multi-Server

**Client Browser (please complete the following information):**
Any

**Additional context**
This problem does not seem to appears with anything other than server pool backed workspaces.

**Comment by j-travis** [comment]
When a user (or admin) deletes a workspace that is a part of a pool, its not deleted immediately. The kasm_manager container for that zone is responsible for looking detecting the workspaces is in a pending delete status and it should delete it. 

It should occur within a few minutes.

Can you look for any errors that may be occurring in the kasm_manager. You can look in the applications logs via the ui or even better tail the logs and repeat the test.

```
 sudo tail -f /opt/kasm/current/log/manager_api_server.log
```
and repeat the test.

You should see something like this

```
2023-07-18 23:13:49,764 [DEBUG] manager_api_server: Provider Manager Initialized
2023-07-18 23:13:49,772 [DEBUG] manager_api_server: Manager (46af4154-5251-4483-a3d7-156026b26e06) checking in
2023-07-18 23:13:49,786 [DEBUG] manager_api_server: I (46af4154-5251-4483-a3d7-156026b26e06) am the primary manager. Processing
2023-07-18 23:13:49,798 [DEBUG] manager_api_server: Provider (hardware) Status: Current Agents (1) , Max Agents (0)
2023-07-18 23:13:49,800 [DEBUG] manager_api_server: Looking for missing servers
2023-07-18 23:13:49,806 [DEBUG] manager_api_server: (0) is now marked as [missing]
2023-07-18 23:13:49,807 [DEBUG] manager_api_server: Looking for dead servers
2023-07-18 23:13:49,812 [DEBUG] manager_api_server: (0) now marked as [dead]
2023-07-18 23:13:49,813 [DEBUG] manager_api_server: Looking for expired kasms
2023-07-18 23:13:49,819 [DEBUG] manager_api_server: Looking for admin_destroyed_pending kasms
2023-07-18 23:13:49,843 [DEBUG] manager_api_server: Sending connection proxy disconnect request to (https://proxy:443/guac/vm_disconnect) for kasm_id (f5743331-859a-4a24-b010-c675b82e4fa7)
2023-07-18 23:13:49,860 [DEBUG] manager_api_server: Updated Accounting record for kasm_id (f5743331-859a-4a24-b010-c675b82e4fa7)
2023-07-18 23:13:49,868 [INFO] manager_api_server: Kasm (f5743331-859a-4a24-b010-c675b82e4fa7) destroyed after (20.868141) seconds of total uptime
2023-07-18 23:13:49,873 [DEBUG] manager_api_server: Function (provider_manager.destroy_kasm) executed in (0.03460860252380371) seconds
```

**Comment by frznfngrs** [comment]
I have looked at the manager logs for both managers that I have.    What I am noticing that is on the manager where server pool sessions delete correctly I see the messages as expected as in your example.

On the manager were server pool sessions do not delete and go into a forever pending state I never see these key lines from your example at all.

> 2023-07-18 23:13:49,843 [DEBUG] manager_api_server: Sending connection proxy disconnect request to (https://proxy:443/guac/vm_disconnect) for kasm_id (f5743331-859a-4a24-b010-c675b82e4fa7)
> 2023-07-18 23:13:49,860 [DEBUG] manager_api_server: Updated Accounting record for kasm_id (f5743331-859a-4a24-b010-c675b82e4fa7)
> 2023-07-18 23:13:49,868 [INFO] manager_api_server: Kasm (f5743331-859a-4a24-b010-c675b82e4fa7) destroyed after (20.868141) seconds of total uptime
> 2023-07-18 23:13:49,873 [DEBUG] manager_api_server: Function (provider_manager.destroy_kasm) executed in (0.03460860252380371) seconds

Those lines never come at all, and no errors at all either.

**Comment by j-travis** [comment]
Can you please post screenshots of your `/#/managers` and `/#/zones` pages

**Comment by frznfngrs** [comment]
<img width="577" alt="image" src="https://github.com/kasmtech/workspaces-issues/assets/19555532/99721bbe-f1ff-462e-ba31-b645a52c6603">

<img width="552" alt="image" src="https://github.com/kasmtech/workspaces-issues/assets/19555532/96914b68-d5b5-4d90-b4e6-5984831f2634">

**Comment by frznfngrs** [comment]
Any further information I can provide?

**Comment by j-travis** [comment]
You might try stopping and starting the services on the problematic webapp server

```
sudo /opt/kasm/bin/stop
sudo /opt/kasm/bin/start
```

I'd be interested to see the logs that are produced by the problematic manager after you delete the session

**Comment by j-travis** [comment]
FYSA, I believe we've identified the bug that may be impacting you.  
Will let you know what we find and when a fix is available.

**Comment by frznfngrs** [comment]
Thank you for the update

**Comment by rickkoliser** [comment]
We have identified and fixed a bug that prevented admin_delete_pending server pool Kasms from being deleted. It will be in our developer preview release in a few days. We did not see a bug with user_delete_pending. I would recommend trying once the fix is in our developer preview to see if you still experience the issue. I will comment on this ticket once the fix is available in the developer preview release.
